[Make] Avoid passing default xcodebuild options to installsrc

emw-apple opened the pull request:
#### c68a647dbadb0275a7e3c30320c8a56b8e08afe1
<pre>
[Make] Avoid passing default xcodebuild options to installsrc
<a href="https://bugs.webkit.org/show_bug.cgi?id=242089">https://bugs.webkit.org/show_bug.cgi?id=242089</a>
rdar://95680722

Reviewed by Alexey Proskuryakov.

Our &quot;standard&quot; xcodebuild options, as computed by `XcodeOptionString()`
in webkitdirs, are not suitable for `xcodebuild installsrc`. In
particular, they can lead to PIFCache directories being created in
WebKitBuild that are created as a side effect of installsrc.

The -alltargets flag added in
<a href="https://bugs.webkit.org/show_bug.cgi?id=220370">https://bugs.webkit.org/show_bug.cgi?id=220370</a> should no longer be
needed, as installsrc is project (not target) based, and we no longer
pass an XCODE_TARGET selection flag.

* Makefile.shared:

Canonical link: <a href="https://commits.webkit.org/252050@main">https://commits.webkit.org/252050@main</a>
</pre>

emw-apple commented on Makefile.shared:
```diff
@@ -154,10 +154,6 @@ else
-else
-	@$(call invoke_xcode,,,installsrc SRCROOT="$(SRCROOT)$(PATH_FROM_ROOT)")
-endif
+	xcodebuild$(if $(XCODE_TARGET), -alltargets) installsrc SRCROOT="$(SRCROOT)$(PATH_FROM_ROOT)"
```
self-nit: I am not sure I understand what `-alltargets` does, since I believe installsrc operates on the entire project. Maybe we don't need it?

aproskuryakov replied:
Not sure. I cc'ed you on an older bug where something related was changed, but I'm not sure if it's actually relevant.

emw-apple replied:
Makes sense. I think it explains why we got this flag: The common Xcode settings pass a `-target` selection when building libwebrtc.xcodeproj for watchOS. Now, since we won't be passing the common options to installsrc, we can omit this.

emw-apple committed:
[Make] Avoid passing default xcodebuild options to installsrc
https://bugs.webkit.org/show_bug.cgi?id=242089
rdar://95680722

Reviewed by Alexey Proskuryakov.

Our "standard" xcodebuild options, as computed by `XcodeOptionString()`
in webkitdirs, are not suitable for `xcodebuild installsrc`. In
particular, they can lead to PIFCache directories being created in
WebKitBuild that are created as a side effect of installsrc.

The -alltargets flag added in
https://bugs.webkit.org/show_bug.cgi?id=220370 should no longer be
needed, as installsrc is project (not target) based, and we no longer
pass an XCODE_TARGET selection flag.

* Makefile.shared:

Canonical link: https://commits.webkit.org/252050@main
